galsim/phase_screens.py

```diff
@@ -284,6 +473,41 @@ def _random_screen(self):
         noise = utilities.rand_arr(self._psi.shape, gd)
         return fft.ifft2(fft.fft2(noise)*self._psi).real
 
+    def _setShare(self):
+        tab2d = LookupTable2D(self._xs, self._ys, self._screen, edge_mode='wrap')
+
+        xshare = np.frombuffer(self._objDict['x'], dtype=np.float64)
+        yshare = np.frombuffer(self._objDict['y'], dtype=np.float64)
+        fshare = np.frombuffer(self._objDict['f'], dtype=np.float64)
+        fshare = fshare.reshape((self.npix+1, self.npix+1))
+        xshare[:] = tab2d.x
+        yshare[:] = tab2d.y
+        fshare[:] = tab2d.f
+        self._objDict['x0'].value = tab2d.x0
+        self._objDict['y0'].value = tab2d.y0
+        self._objDict['xperiod'].value = tab2d.xperiod
+        self._objDict['yperiod'].value = tab2d.yperiod
+        # Need to reset this here in case we did a boiling update
+        self._tab2d = _LookupTable2D(
+            xshare, yshare, fshare, 'linear', 'wrap', 0.0,
+            x0=self._objDict['x0'].value, y0=self._objDict['y0'].value,
+            xperiod=self._objDict['xperiod'].value, yperiod=self._objDict['yperiod'].value
+        )
```

Comment:
Could avoid some code duplication between this and `_tab2d` below by removing this `self._tab2d = ` bit.  (Maybe also adding a `del self._tab2d` in case boiling update.)  I think the lazy_property will make it when needed in the main process as well as any sub-processes.  

Would this be inefficient?  Or any other problem?

Reply:
Sounds good.  I actually tried deleting this line earlier, but put it back b/c of the boiling update.  I hadn't thought of `del self._tab2d`, which seems like a more elegant solution.